packages/amplify-appsync-simulator/src/server/index.ts

```diff
@@ -2,27 +2,57 @@ import { OperationServer } from './operations';
 import { SubscriptionServer } from './subscription';
 import { AmplifyAppSyncSimulator } from '..';
 import { AppSyncSimulatorServerConfig } from '../type-definition';
+import { Server, createServer } from 'http';
+import portFinder from 'portfinder';
+import e2p from 'event-to-promise';
+import { address as getLocalIpAddress } from 'ip';
+import { AppSyncSimulatorSubscriptionServer } from './websocket-subscription';
+
+const BASE_PORT = 8900;
+const MAX_PORT = 9999;
 
 export class AppSyncSimulatorServer {
   private operationServer: OperationServer;
   private subscriptionServer: SubscriptionServer;
+  private httpServer: Server;
+  private realTimeSubscriptionServer: AppSyncSimulatorSubscriptionServer;
 
-  constructor(config: AppSyncSimulatorServerConfig, simulatorContext: AmplifyAppSyncSimulator) {
+  private _url: string;
+  constructor(private config: AppSyncSimulatorServerConfig, private simulatorContext: AmplifyAppSyncSimulator) {
     this.subscriptionServer = new SubscriptionServer(config, simulatorContext);
     this.operationServer = new OperationServer(config, simulatorContext, this.subscriptionServer);
+    this.httpServer = createServer(this.operationServer.app);
+    this.realTimeSubscriptionServer = new AppSyncSimulatorSubscriptionServer(simulatorContext, this.httpServer, '/graphql');
   }
-  async start() {
+
+  async start(): Promise<void> {
     await this.subscriptionServer.start();
     await this.operationServer.start();
+    await this.realTimeSubscriptionServer.start();
+
+    let port = this.config.port;
```

Comment:
When a port in config comes in we should still run a portfinder on that one to see if it is free or not, in case if it is not free, we can print a warning for customers that we picked a different port from the configured one.